src/tests/src/solver/modelParser/testModelTranslator.cpp

```diff
@@ -33,36 +32,39 @@
 
 using namespace Antares::Solver;
 
-// Test empty library
-BOOST_AUTO_TEST_CASE(Empty_library_is_valid)
+struct Fixture
 {
     ModelParser::Library library;
-    ObjectModel::Library lib = ModelConverter::convert(library);
+    Registry<Nodes::Node> registry;
+};
+
+// Test empty library
+BOOST_FIXTURE_TEST_CASE(Empty_library_is_valid, Fixture)
+{
+    ObjectModel::Library lib = ModelConverter::convert(library, registry);
```

Comment:
```C++
ObjectModel::Library lib = ModelConverter::convert(library, registry);
```
Here we have a parsed library, and we want to convert it into an object Library. 
So we a source and a target.
Why would we need a **registry** here ?

Ok, it's used in the depth of the code to build tree / expressions (associated to variables , constraints and objective) form **std::string**. But do we have to pass it all the way down to where needed ?
The **registry** should be hidden and created either : 
- to where it's really needed (in classes **Variables**, **Constraints**, ...)
- directly in the top node of the AST tree it's supposed be associated with

But not here, in my opinion.
Here we should have : 
```C++
ObjectModel::Library lib = ModelConverter::convert(library);
```

Reply:
maybe hide one Registry instance inside each Expression instance ?

Reply:
See #2501 for a fix